neptune/internal/backends/hosted_neptune_backend.py

```diff
@@ -915,7 +920,53 @@ def _create_authenticator(self, api_token, ssl_verify, proxies):
     @with_api_exceptions_handler
     def _create_client_config(self, api_token, backend_client):
         config = backend_client.api.getClientConfig(X_Neptune_Api_Token=api_token).response().result
-        return ClientConfig(api_url=config.apiUrl, display_url=config.applicationUrl)
+        
+        min_recommended = None
+        min_compatible = None
+        max_compatible = None
+
+        if hasattr(config, "versions"):
```

Comment:
else section with a warning would be nice - non-SaaS users would get a warning with information that this client is newer than they (propably) support